cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -114,6 +121,40 @@ protected void startUp() throws Exception {
     runningThread = Thread.currentThread();
   }
 
+  private void checkSuspended() {
+    lock.lock();
+    try {
+      while (isSuspended) {
+        condition.await();
+      }
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  public void suspend() throws Exception {
```

Comment:
Yes. suspend happens only after all the current steps are executed. Added javadoc.